relay-otel/src/lib.rs

```diff
@@ -0,0 +1,240 @@
+)]
+
+use opentelemetry_proto::tonic::common::v1::any_value::Value as OtelValue;
+use relay_event_schema::protocol::{Attribute, AttributeType};
+use relay_protocol::Value;
+
+/// Converts an OpenTelemetry AnyValue to a Sentry attribute.
+///
+/// This function handles the conversion of OpenTelemetry attribute values to Sentry attribute types.
+/// Complex types like arrays and key-value lists are serialized to JSON strings for safety and
+/// compatibility.
+///
+/// For array and key-value list values, this function filters out nested complex types
+/// (nested arrays and key-value lists) before serialization to prevent issues with
+/// the OTLP protocol and ensure safe handling.
+pub fn otel_value_to_attribute(otel_value: OtelValue) -> Option<Attribute> {
+    let (ty, value) = match otel_value {
+        OtelValue::StringValue(s) => (AttributeType::String, Value::String(s)),
+        OtelValue::BoolValue(b) => (AttributeType::Boolean, Value::Bool(b)),
+        OtelValue::IntValue(i) => (AttributeType::Integer, Value::I64(i)),
+        OtelValue::DoubleValue(d) => (AttributeType::Double, Value::F64(d)),
+        OtelValue::BytesValue(bytes) => {
+            let s = String::from_utf8(bytes).ok()?;
+            (AttributeType::String, Value::String(s))
+        }
+        OtelValue::ArrayValue(array) => {
+            // Filter out nested arrays and key-value lists for safety.
+            // This is not usually allowed by the OTLP protocol, but we filter
+            // these values out before serializing for robustness.
+            let safe_values: Vec<serde_json::Value> = array
+                .values
+                .into_iter()
+                .filter_map(|v| match v.value? {
+                    OtelValue::StringValue(s) => Some(serde_json::Value::String(s)),
+                    OtelValue::BoolValue(b) => Some(serde_json::Value::Bool(b)),
+                    OtelValue::IntValue(i) => {
+                        Some(serde_json::Value::Number(serde_json::Number::from(i)))
+                    }
+                    OtelValue::DoubleValue(d) => {
+                        serde_json::Number::from_f64(d).map(serde_json::Value::Number)
+                    }
+                    OtelValue::BytesValue(bytes) => {
+                        String::from_utf8(bytes).ok().map(serde_json::Value::String)
+                    }
+                    // Skip nested complex types for safety
+                    OtelValue::ArrayValue(_) | OtelValue::KvlistValue(_) => None,
+                })
+                .collect();
+
+            // Serialize the array values as a JSON string. Even though there is some nominal
+            // support for array values in Sentry, it's not robust and not ready to be used.
+            // Instead, serialize arrays to a JSON string, and have the UI decode the JSON if
+            // possible.
+            let json = serde_json::to_string(&safe_values).unwrap_or_default();
```

Comment:
I assume we don't want to set the attribute if it cannot be stringified?

```suggestion
            let json = serde_json::to_string(&safe_values).ok()?;
```

Reply:
yup for sure, changed.